backends/ebpf/ebpftargets.py

```diff
@@ -50,116 +51,140 @@ def nextWord(text, sep=None):
+        print("***", file=err_file, *message)
+        err_file.close()
+
+
+def report_output(file, verbose, *message):
+    if (verbose):
+        print(file=sys.stdout, *message)
+
+    if (file and file != sys.stdout):
+        out_file = open(file, "a+")
+        print("", file=out_file, *message)
+        out_file.close()
+
 
-    if local.process.returncode != SUCCESS:
-        procstderr = open(stderr, "r")
-        errtext = procstderr.read()
-        procstderr.close
-    return local.process.returncode, errtext
+def run_timeout(options, args, timeout, outputs, errmsg):
+    report_output(outputs["stdout"],
+                  options.verbose, "Executing ", " ".join(args))
+    proc = None
 
+    def kill(process):
+        process.kill()
 
-def comparePacket(expected, received):
-    received = ''.join(ByteToHex(str(received)).split()).upper()
+    if outputs["stderr"] is not None:
+        try:
+            proc = Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        except OSError as e:
+            report_err(outputs["stderr"], "Failed executing: ", e)
+    if proc is None:
+        # Never even started
+        report_err(outputs["stderr"], "Process failed to start")
+        return FAILURE
+
+    timer = Timer(TIMEOUT, kill, [proc])
+    try:
+        timer.start()
+        out, err = proc.communicate()
+    finally:
+        timer.cancel()
+    msg = ("\n########### PROCESS OUTPUT BEGIN:\n"
+           "%s########### PROCESS OUTPUT END\n" % out)
+    report_output(outputs["stdout"], options.verbose, msg)
+    if proc.returncode != SUCCESS:
+        report_err(outputs["stderr"], "Error %d: %s\n%s" %
+                   (proc.returncode, errmsg, err))
+    return proc.returncode
+
+
+def compare_pkt(outputs, expected, received):
+    received = ''.join(byte_to_hex(str(received)).split()).upper()
     expected = ''.join(expected.split()).upper()
     if len(received) < len(expected):
-        reportError("Received packet too short",
-                    len(received), "vs", len(expected))
+        report_err(outputs["stderr"], "Received packet too short",
+                   len(received), "vs", len(expected))
         return FAILURE
     for i in range(0, len(expected)):
         if expected[i] == "*":
             continue
         if expected[i] != received[i]:
-            reportError("Received packet ", received)
-            reportError("Packet different at position", i,
-                        ": expected", expected[i], ", received", received[i])
-            reportError("Full received packed is ", received)
+            report_err(outputs["stderr"], "Received packet ", received)
+            report_err(outputs["stderr"], "Packet different at position", i,
+                       ": expected", expected[i], ", received", received[i])
+            report_err(outputs["stderr"], "Full received packed is ", received)
             return FAILURE
     return SUCCESS
 
 
+def require_root(outputs):
```

Comment:
How are people supposed to run this test with root permissions?
They should not run all tests with root permissions.